pkg/utils/testutil/testutil.go

```diff
@@ -101,3 +103,24 @@ func InitTempFileLogger(level string) (fname string) {
 	log.ReplaceGlobals(lg, p)
 	return fname
 }
+
+// GenerateTestDataConcurrently generates test data concurrently.
+func GenerateTestDataConcurrently(count int, f func(int)) {
+	var wg sync.WaitGroup
+	tasks := make(chan int, count)
+	workers := runtime.NumCPU()
+	for w := 0; w < workers; w++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for i := range tasks {
+				f(i)
+			}
+		}()
+	}
+	for i := 0; i < count; i++ {
+		tasks <- i
+	}
```

Comment:
Why not just pass the `i` as a goroutine function parameter above?

Reply:
Because we need to divide tasks to different goroutines